### PR TITLE
[RLlib] Experiment with py_func as a means to further unify tf and torch (Schedule classes).

### PR DESCRIPTION
**Pull request by sven1977**
Helps with the push to unify handling of tf and torch in different DL components such that Algorithm developers will have it easier to translate from one framework to the other. 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: rllib/utils/schedules/schedule.py [hunk 1]
```diff
@@ -26,19 +29,26 @@ def __init__(self, framework=None):
...
-            t (int): The time value (e.g. a time step).
-                NOTE: This could be a tf.Tensor.
+            t (int): The time step. This could be a tf.Tensor.
 
         Returns:
             any: The calculated value depending on the schedule and `t`.
         """
         raise NotImplementedError
 
+    def value(self, t):
```
Comment (ericl) [inline rllib/utils/schedules/schedule.py]: I think we can get rid of self.framework variable by renaming this call to .tf_value right? That would also be more explicit.

Comment (sven1977) [reply rllib/utils/schedules/schedule.py]: Could we leave it as is for now? There are a lot of possible drawbacks to this split method approach. Once we are done with the new Exploration classes, we can revisit this question and see what works and what doesn't.

Comment (sven1977) [reply rllib/utils/schedules/schedule.py]: RLlib tests all seem ok.

**Review by ericl — APPROVED** [review]

**Comment by sven1977** [comment]
Thanks.

**Pull request merged**